libgstc/pygstc/gstc.py

```diff
@@ -0,0 +1,788 @@
+        self._send_cmd_line(['delete', uri, name])
+
+    def element_get(
+        self,
+        pipe_name,
+        element,
+        prop,
+    ):
+        """
+        Queries a property in an element of a given pipeline.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        element: string
+            The name of the element
+        prop: string
+            The name of the property
+
+        Returns
+        -------
+        result : string
+            Command response
+        """
+
+        self._logger.info(
+            'Getting value of element %s %s property in pipeline %s' %
+            (element, prop, pipe_name))
+        result = self._send_cmd_line(cmd_line=['element_get',
+                                               pipe_name, element, prop])
+        return result['response']['value']
+
+    def element_set(
+        self,
+        pipe_name,
+        element,
+        prop,
+        value,
+    ):
+        """
+        Sets a property in an element of a given pipeline.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        element: string
+            The name of the element
+        prop: string
+            The name of the property
+        value: string
+            The value to set
+        """
+
+        self._logger.info('Setting element %s %s property in pipeline %s to:%s'
+                          % (element, prop, pipe_name, value))
+        self._send_cmd_line(['element_set', pipe_name, element, prop,
+                             value])
+
+    def event_eos(self, pipe_name):
+        """
+        Send an end-of-stream event.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        """
+
+        self._logger.info('Sending end-of-stream event to pipeline %s'
+                          % pipe_name)
+        self._send_cmd_line(['event_eos', pipe_name])
+
+    def event_flush_start(self, pipe_name):
+        """
+        Put the pipeline in flushing mode.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        """
+
+        self._logger.info('Putting pipeline %s in flushing mode'
+                          % pipe_name)
+        self._send_cmd_line(['event_flush_start', pipe_name])
+
+    def event_flush_stop(self, pipe_name, reset='true'):
+        """
+        Take the pipeline out from flushing mode.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        reset: string
+            Reset the event flush: 'true' or 'false'
+        """
+
+        self._logger.info('Taking pipeline %s out of flushing mode'
+                          % pipe_name)
+        self._send_cmd_line(['event_flush_stop', pipe_name, reset])
+
+    def event_seek(
+        self,
+        pipe_name,
+        rate='1.0',
```

Comment:
From the python perspective, these are doubles and integers. It's strange to receive strings.

Reply:
apply this retroactively to the rest of the methods.